Update IASI, ATMS and Add CrIS. and update all Python scripts to follow pycodestye

emilyhcliu opened the pull request:
**This PR updates two things:**
- Update mapping file and Python script for IASI, CrIS , ATMS 

   Update the following to IASI:**
   1. Rename mtiasi to iasi
   2. Change logging to self.log.warning
   3. Add map_path function

   Add new data type - cris-fsr**

   Update the following for ATMS:**
   1. Change logging to self.log.warning
   2. Add map_path function

- Add Python coding norm check (pycodestyle) --- this is an initial implementation so we 
   have a coding norm check for Python script. This will be updated to use pytest in another PR
   To use it:
   - Load obsforge modules (~eliu/modules/env_obsforge.sh)
   - go to ./sorc/spoc/build
   - ctest -VV -R spoc_python_coding_norms

- Modify all Python scripts to follow pycodestyle (ctest to be added in another PR soon)
   (No change in science)

emilyhcliu committed:
Update IASI

emilyhcliu committed:
change Mtiasi to Iasi

PraveenKumar-NOAA reviewed (approved):
Looks good to me.

rmclaren reviewed (commented):
Line 19 bufr.Parser ... Shouldn't it be self.mapping_path ?

Or just do:
container = super().make_obs(comm, input_path)

emilyhcliu committed:
Add map_path and super().make_obs

emilyhcliu commented:
> Line 19 bufr.Parser ... Shouldn't it be self.mapping_path ?
> 
> Or just do: container = super().make_obs(comm, input_path)

I thought the `map_path` function has be default part of the obs_builder, so I removed it.
OK.  I added map_path back and changed to get container with super().make_obs....

emilyhcliu committed:
Add mapping file and python script for CrIS

emilyhcliu committed:
add map_path function and change logging to self.log.warning

emilyhcliu committed:
add python coding norm ctest

emilyhcliu committed:
Merge branch 'feature/ctest' into feature/obs_builder_iasi

emilyhcliu committed:
Modify to pass pycodestyle check

emilyhcliu committed:
fix coding norm

nicholasesposito reviewed (approved):
Looks good.

emilyhcliu commented:
The atms, iasi and cris worked as expected.
I am testing the satwind now.  It is not all working yet.  
I am working on it.  I will ask you to review again after I make the Satwind converters work.  Sorry about that!

emilyhcliu commented:
I will make a table for satwind here and check the data type when I am done testing:
- [ ] abi
- [ ] seviri
- [ ] ahi
- [ ] leogeo
- [ ] avhrr
- [ ] viirs
- [ ] modis

emilyhcliu commented:
> I will make a table for satwind here and check the data type when I am done testing:
> 
> * [ ]  abi
> * [ ]  seviri
> * [ ]  ahi
> * [ ]  leogeo
> * [ ]  avhrr
> * [ ]  viirs
> * [ ]  modis

I am going to test satwind in another PR.  ([Issue #33](https://github.com/NOAA-EMC/spoc/issues/33) opened)
So, this PR includes:
1. add CrIS and update IASI, and ATMS 
2. make sure all Python scripts uner spoc/dump/mapping follow pycodestyle.

emilyhcliu commented:
@nicholasesposito @PraveenKumar-NOAA @rmclaren 
This PR is ready for review.

Summary of this PR
- Update ATMS and IASI
- Add CrIS (new)
- Add capability of do ctest for checking coding norm (pycodestyle) so we can at least check the python code when we add Python scripts to SPOC.  This pycodestyle check will be replaced by pytest in the future
- modify all Python scripts to satisfy pycodestyle --- most of the changes are fixing white spaces and spacing between Classes and Functions; no science change ( do not change results)

emilyhcliu committed:
Update satwind YAML and Python scripts (#34)

This PR test and update satellite AMV converters
Please see details documented in Issue #33 

- a few bug fixes
- Change `qualityInformationWithoutForecast` to `qiWithoutForecast`

All Python scripts passed coding norm check

emilyhcliu commented:
@rmclaren  I saw that you merged PR #34 into this PR already.  Thanks.
I think this PR is ready to merge too!
Thanks.
